### ISSUE TITLE
Facing microphone issues in the latest teams-for-linux version '1.4.1'

### ISSUE THREAD

**Issue by emailstorbala** [IsmaelMartinez/teams-for-linux#1065]
**Describe the bug**
When using the latest version of 'teams-for-linux', the microphone audio is malfunctioning.

**To Reproduce**
Steps:
1. Upgrade to latest version 1.4.1
2. Connect the teams and try a meeting with a separate microphone device.
3. The other meeting persons are receiving an echo/distorted voices.

**Expected behavior**
Clear audio should be received by other recipients.

**Screenshots**
Since Audio is the issue, screenshot is not applicable.

**
Laptop: Thinkpad T470
OS: Debian 12 (Testing)
DE: KDE/Wayland

**Additional context**
Tried multiple times joining the same meeting. Restarted Teams also. No impact.

Only after going back to the older version of teams-for-linux '1.3.26', the issue is getting resolved.

**Comment by emailstorbala** [comment]
Now I am facing the same issue in 1.3.26 version also.

**Comment by fran-penedo** [comment]
I can confirm the issue in 1.3.26 and 1.4.1.1, @emailstorbala please reopen.

**Comment by pacoarribas** [comment]
Same issue, with Thinkpad T490, Dell Latitude, Pop! OS

**Comment by IsmaelMartinez** [comment]
Can you check if this is the same issue using the web browser? Sound issues normally are related to system configuration and/or MS changes.

**Comment by IsmaelMartinez** [comment]
Otherwise, can you check if using the `disableAutogain` option does the trick? 

https://github.com/IsmaelMartinez/teams-for-linux/blob/develop/app/config/README.md

**Comment by sisco70** [comment]
Same issue. Ubuntu 22.04 LTS.  Teams for Linux version 1.4.1.

The meeting participants heard an echo. I closed Teams for Linux, connected to the meeting from Chrome, and everything settled down. I did not make any changes to the audio configuration.

I have not yet been able to try version 1.4.2

**Comment by Mushoz** [comment]
I am facing the exact same issue. It was very difficult for our customer to follow me in our Teams meeting. I am using the browser version for now to work around this issue.

**Comment by IsmaelMartinez** [comment]
Did you try the disable autogain?

**Comment by sisco70** [comment]
I just made a voice call with version 1.4.2 and there were no Echo problems, without having to change the flag autogain.

I don't have meetings today to test (in case it behaves differently)

**Comment by gpcmol** [comment]
I also had the same issue. My (unproven) guess is that after the update to 1.4.1 (last Friday on my Lubuntu OS/computer), teams adjusted the microphone to 100%.
After some fiddling, I put it to 24% and my collegues are nice to me again.
![image](https://github.com/IsmaelMartinez/teams-for-linux/assets/1242709/dd5c6231-b479-4ca9-9d2c-956c6445a375)

**Comment by Danipiario** [comment]
Same issue for me, even with disableAutogain flag.
I've tried version 1.4.1, 1.4.2 and also 1.4.3 from edge channel with no luck.
It seems to be releated to Microsoft Teams New version.
Unfortunately the issue is still present with teams on browser for me

**Comment by IsmaelMartinez** [comment]
sorry to hear that. unfortunately if it is also happening in the browser, then is a MS issue. Fortunately, they got 100s more developers than us.

**Comment by dha4171** [comment]
I too get the echo with the app and with the browser.  I just tested the disableAutogain and it seems to work now.  First call was always ok.  Subsequent calls would have the echo.  I will definitely be using it more tomorrow and will report back.

**Comment by gpcmol** [comment]
After I posted the screenshot above, I thought the issue was resolved, but it seems that the echo is back unfortunately

**Comment by gpcmol** [comment]
https://answers.microsoft.com/en-us/msteams/forum/all/people-can-hear-an-echo-when-using-teams-so-not/52cbda27-6494-47a4-904b-51db20345a76?page=5

**Comment by Danipiario** [comment]
I reinstall the snap and today it's working

**Comment by IsmaelMartinez** [comment]
I will put the 'won't fix' tag but leave it open in case there is any workaround that can be shared. Unfortunately this looks like a MS issue.

**Comment by emailstorbala** [comment]
I am using version 1.4.2 (teams-for-linux) from yesterday with  the flag '--disableAutogain'. Its working without issue till today morning.

Today there was an update 1.4.3 (with some wayland screen sharing rework). This brought in a new issue while sharing the desktop. During desktop sharing, the audio is completely distorted/ too much echo. This is tested with and without the flag '--disableAutogain'.

Now I have reverted back to version 1.4.2 with the flag '--disableAutogain' and its back to normal.

CC: @IsmaelMartinez

**Comment by emailstorbala** [comment]
> After I posted the screenshot above, I thought the issue was resolved, but it seems that the echo is back unfortunately

@gpcmol If we don't set --disableAutogain, the microphone volume automatically raising to max/full.( atleast < 1.4.2). In 1.4.3, with --disableAutogain, it works good without screen sharing.

**Comment by IsmaelMartinez** [comment]
Can you try removing the cache? I would be surprised at the desktop sharing changes affecting this, as I don't think we share the device sound.

**Comment by gpcmol** [comment]
@emailstorbala I did the same now and the echo is gone (for how long ;)?)
My settings are:
1.4.2
--disableAutogain
input device on 24% (volume control Lubuntu)
teams Device settings: Noise suppression = High

**Comment by tsadan** [comment]
I looks like in 1.4.3 another input device starts, when starting to share a screen:
Input device without sharing:
![image](https://github.com/IsmaelMartinez/teams-for-linux/assets/1706347/b72bb6ba-a439-4326-ac67-a68a3eb56b63)
Input devices right after start to share:
![image](https://github.com/IsmaelMartinez/teams-for-linux/assets/1706347/0f78fa64-0d59-4b21-a7a5-eed966b207e1)
With that second input enabled, meeting paticipiants do hear echos. When the second input device is muted in PA volume control, sound is ok.
Can the start of the second input device be avoided somehow ?

**Comment by IsmaelMartinez** [comment]
I wonder if that screensharing is trying to share your computer audio... it should not do that by default but that is my guess.

**Comment by enzy** [comment]
The issue occurs for me when I start a screen share. From that point, others hear me twice (like an echo).

**Comment by tsadan** [comment]
As I read [here](https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v1.4.3), the wayland screenshare part has been reworked. Maybe now there are more sources grabed than needed, not only the "screen" but more like audio devices.

**Comment by tsadan** [comment]
@IsmaelMartinez What do you think about removing the "wontfix" label, as it is (at least from what I read) confirmed to be related to wayland displaysharing and is about to be fixed by @herbiejhopkins ?

**Comment by IsmaelMartinez** [comment]
this might be fixed under https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v1.4.5

**Comment by MonstrousOgre** [comment]
I was facing the issue while screen sharing yesterday. It seems to be fixed after I updated to 1.4.5

**Comment by damien-louis** [comment]
Thanks @IsmaelMartinez. Do you know when v1.4.5 will be released?

**Comment by jijojosephk** [comment]
@damien-louis `1.4.6` is just released. `flatpak` version is being built and gonna be available soon.

**Comment by jijojosephk** [comment]
publish fails seems like a problem with flathub.

```terminal
flatpak run --command=flat-manager-client org.flatpak.Builder purge https://hub.flathub.org/api/v1/build/79008
 in dir /var/lib/buildbot-master/master/workers/MasterWorker8/publish/build (timeout 1200 secs)
 watching logfiles {}
 argv: [b'flatpak', b'run', b'--command=flat-manager-client', b'org.flatpak.Builder', b'purge', b'https://hub.flathub.org/api/v1/build/79008']
 using PTY: True
Purging build https://hub.flathub.org/api/v1/build/79008
Api call to https://hub.flathub.org/api/v1/build/79008/purge failed with status 400, details: {'error-type': 'generic-error', 'message': "Can't prune build while in use", 'status': 400}
program finished with exit code 1
elapsedTime=0.349693
```

**Comment by jijojosephk** [comment]
![image](https://github.com/IsmaelMartinez/teams-for-linux/assets/11662148/8207da14-aca1-4428-bb3c-cd1840c09f8e)

**Comment by jijojosephk** [comment]
It appears that there's a new permission workflow added to the flathub builds. There was a recent change in permission to flatpak manifest which needs to be approved by admins. @bbhtt would do it.

https://github.com/flathub/com.github.IsmaelMartinez.teams_for_linux/pull/83#issuecomment-1913530915

**Comment by jijojosephk** [comment]
![image](https://github.com/IsmaelMartinez/teams-for-linux/assets/11662148/d0296b26-645a-43ed-9ee7-bf11196fe6de)

**Comment by IsmaelMartinez** [comment]
I think they just approved
